log4j-core-test/src/test/java/org/apache/logging/log4j/core/config/plugins/util/PluginManagerTest.java

```diff
@@ -0,0 +1,224 @@
+            assertThat(message.getLevel()).isEqualTo(Level.WARN);
+            assertThat(message.getMessage().getFormattedMessage())
+                    .as("Status logger message")
+                    // The message specifies which plugin is not registered
+                    .contains("customplugin.FixedStringLayout")
+                    // The message provides a link to the registration instructions
+                    .contains("https://logging.apache.org/log4j/2.x/manual/plugins.html#plugin-registry");
+        });
+    }
+
+    @Test
+    @UsingStatusListener
+    void missing_plugin_descriptor_issues_warning(final ListStatusListener listener) {
+        final ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread()
+                    .setContextClassLoader(new FilteringClassLoader(getClass().getClassLoader(), null));
+            assertThat(LoaderUtil.getClassLoader().getResource(PLUGIN_DESCRIPTOR))
+                    .isNull();
+            final PluginManager manager = new PluginManager(Node.CATEGORY);
+            manager.collectPlugins(null);
+            assertThat(listener.findStatusData(Level.WARN)).anySatisfy(message -> {
+                assertThat(message.getLevel()).isEqualTo(Level.WARN);
+                assertThat(message.getMessage().getFormattedMessage())
+                        .as("Status logger message")
+                        // The message provides a link to plugin descriptor troubleshooting.
+                        .contains("https://logging.apache.org/log4j/2.x/faq.html#plugin-descriptors");
+            });
+        } finally {
+            Thread.currentThread().setContextClassLoader(tccl);
+        }
+    }
+
+    /**
+     * Tests if removing descriptors of standard Log4j artifacts and replacing them with a custom one issues warnings.
+     * <p>
+     *     This often happens in shading scenarios.
+     * </p>
+     */
+    @Test
+    @UsingStatusListener
+    void corrupted_plugin_descriptor_issues_warning(final ListStatusListener listener) throws IOException {
+        // Create empty descriptor
+        final Path customDescriptor = Files.createTempFile("Log4j2Plugins", ".dat");
+        final URL descriptorUrl = customDescriptor.toUri().toURL();
+        final ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        try {
+            // Generate single plugin
+            final PluginEntry entry = new PluginEntry();
+            entry.setKey("CustomPlugin");
+            entry.setClassName("com.example.CustomPlugin");
+            entry.setName("CustomPlugin");
+            entry.setCategory(Node.CATEGORY);
+            // Generate descriptor with single plugin
+            final PluginCache customCache = new PluginCache();
+            customCache.getCategory(Node.CATEGORY).put("CustomPlugin", entry);
+            customCache.writeCache(Files.newOutputStream(customDescriptor));
```

Comment:
`CustomPlugin` should not exist right? If so, would you mind renaming it to `com.example.NotExistingPlugin`, please? (The case is not self-explanatory to those unfamiliar with the test.)

Reply:
It doesn't really matter if the class exists or not.

I added a comment in https://github.com/apache/logging-log4j2/pull/2836/commits/d2bdd02ca9add9cc155d7c8a65f1335bf8d5dd70.